### ISSUE TITLE
Model.findOne(undefined).populate('some-1-to-1-mode') returns a record

### ISSUE THREAD

**Issue by Clan-Utility** [balderdashy/waterline#860]
Model.findOne(undefined) throws Error: .findOne() requires a criteria.

But Model.findOne(undefined).populate('some-1-to-1-mode') returns a record (in my case, the record with id = 1)

The current solution is to always use Model.findOne({id: undefined})

**Comment by devinivy** [comment]
Which adapter are you using?

**Comment by devinivy** [comment]
@Clan-Utility mind sharing which adapter you're using?

**Comment by dmarcelino** [comment]
I've checked the docs, the adapter [findOne.test.js](https://github.com/balderdashy/waterline-adapter-tests/blob/master/interfaces/semantic/findOne.test.js) and the waterline core [query.findOne.js](https://github.com/balderdashy/waterline/blob/master/test/unit/query/query.findOne.js) test and there is nothing suggesting one can search for `undefined`. I don't think this was foreseen. The [`.findOne()`](https://github.com/balderdashy/waterline/blob/master/lib/waterline/query/finders/basic.js#L28) does a lot of criteria normalisation but I didn't dive into that...

@Clan-Utility what is your use case for this?

**Comment by devinivy** [comment]
I think the issue at hand is that, when using populate, you should still receive an error.  Undefined certainly can't be valid criteria in `findOne`.

**Comment by dmarcelino** [comment]
We could replace the [following lines](https://github.com/balderdashy/waterline/blob/master/lib/waterline/query/finders/basic.js#L56-L60):
```javascript
    // If there was something defined in the criteria that would return no results, don't even
    // run the query and just return an empty result set.
    if(criteria === false) {
      return cb(null, null);
    }
```
for
```javascript
    if(criteria === false || criteria === undefined) {
      return cb(null, null);
    }
```

**Comment by dmarcelino** [comment]
This is fixed by PR #972.
